File: doc-examples/example-protobuf-java/src/main/java/example/Telemetry.java
package example;

import io.micronaut.serde.annotation.Serdeable;
import io.micronaut.serde.protobuf.annotation.ProtoField;
import io.micronaut.serde.protobuf.annotation.ProtoType;

// tag::clazz[]
@Serdeable
public record Telemetry(
    String deviceId,                                                    // <1>
    long recordedAt,
    @ProtoField(type = ProtoType.SINT32) int temperatureMilliCelsius
) {
}
// end::clazz[]

File: doc-examples/example-protobuf-kotlin/src/main/kotlin/example/Telemetry.kt
package example

import io.micronaut.serde.annotation.Serdeable
import io.micronaut.serde.protobuf.annotation.ProtoField
import io.micronaut.serde.protobuf.annotation.ProtoType

// tag::clazz[]
@Serdeable
data class Telemetry(
    val deviceId: String,                                                    // <1>
    val recordedAt: Long,
    @field:ProtoField(type = ProtoType.SINT32) val temperatureMilliCelsius: Int
)
// end::clazz[]
